pkg/sql/sqlbase/index_encoding.go

```diff
@@ -270,26 +271,17 @@ func MakeFullKeyFromEncDatums(
 		return nil, err
 	}
 
-	primaryIndex := tableDesc.PrimaryIndex
-	extraColumns := index.ExtraColumnIDs
-	// PrimaryImplicitIdxs maps the position of the implicit column in the key
-	// to the appropriate column in the primary index.
-	primaryImplicitIdxs := make([]int, len(extraColumns))
-	for i, id := range extraColumns {
-		for j, primaryID := range primaryIndex.ColumnIDs {
-			if id == primaryID {
-				primaryImplicitIdxs[i] = j
-			}
-		}
-	}
-	implicitDirs := make([]IndexDescriptor_Direction, len(primaryImplicitIdxs))
-	for i, idx := range primaryImplicitIdxs {
-		implicitDirs[i] = primaryIndex.ColumnDirections[idx]
-	}
-	implicitTypes := make([]ColumnType, len(extraColumns))
-	for i, id := range extraColumns {
-		implicitTypes[i] = tableDesc.ColumnTypes()[id]
+	implicitTypes := make([]ColumnType, len(implicitValues))
+	columnTypes := tableDesc.ColumnTypes()
+	for i := range implicitValues {
+		colID := index.ExtraColumnIDs[i]
+		colIdx := colIdxMap[colID]
+		implicitTypes[i] = columnTypes[colIdx]
 	}
+
+	// Implicit key columns are all sorted ascending (the zero value).
+	implicitDirs := make([]IndexDescriptor_Direction, len(implicitValues))
```

Comment:
Ditto for this - an extra allocation per lookup. A couple of other similar methods accept `nil` for their `directions` field for this same reason (presumably). I'd recommend making that change to `appendEncDatumsToKey` to avoid having to specify this slice.